pandas/io/pytables.py

```diff
@@ -2353,16 +2353,48 @@ def set_atom(self, block, data_converted, use_str: bool):
             # set as a data block
             self.set_atom_data(block)
 
-    def get_atom_string(self, shape, itemsize):
+    @classmethod
+    def _get_atom(cls, values: Union[np.ndarray, ABCExtensionArray]) -> "Col":
+        """
+        Get an appropriately typed and shaped pytables.Col object for values.
+        """
+
+        dtype = values.dtype
+        itemsize = dtype.itemsize
+
+        shape = values.shape
+        if values.ndim == 1:
+            # EA, use block shape pretending it is 2D
+            shape = (1, values.size)
+
+        if is_categorical_dtype(dtype):
+            codes = values.codes
+            atom = cls.get_atom_data(shape, kind=codes.dtype.name)
+        elif dtype.kind == "M":
+            atom = cls.get_atom_datetime64(shape)
+        elif dtype.kind == "m":
```

Comment:
i would prefer we use the is_* methods for this it is much more readable

Reply:
sure